### PR TITLE
Local Cache implementation

### PR DESCRIPTION
**Pull request by MiguelMoll**

### CODE REVIEW

File: provider/local/cache.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"fmt"
+	"time"
+
+	lcache "github.com/convox/praxis/cache"
+	"github.com/convox/praxis/types"
+)
+
+var cacheBuffer = make(map[string]interface{})
```
Comment (ddollar) [inline provider/local/cache.go]: what is this for?

File: provider/local/cache.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package local
+
+import (
+	"fmt"
+	"time"
+
+	lcache "github.com/convox/praxis/cache"
```
Comment (ddollar) [inline provider/local/cache.go]: can this just be `cache` ?

File: types/provider.go [hunk 3]
```diff
@@ -20,6 +20,9 @@ type Provider interface {
 	BuildList(app string) (Builds, error)
 	BuildUpdate(app, id string, opts BuildUpdateOptions) (*Build, error)
 
+	CacheFetch(app, cache, key string) (map[string]string, error)
```
Comment (ddollar) [inline types/provider.go]: Should Cache just store/fetch []byte or string?

Comment (MiguelMoll) [inline types/provider.go]: Definitely could. I was mimicking, Queue API. Open to suggestions here.

File: provider/local/cache.go (hunk 1 above)
Comment (MiguelMoll) [inline provider/local/cache.go]: Forgot to remove it.

File: provider/local/cache.go (hunk 2 above)
Comment (MiguelMoll) [inline provider/local/cache.go]: I renamed the package as I preferred to have a parameter named `cache`.

**Comment by MiguelMoll** [comment]
[local cache implementation](https://trello.com/c/PCpnIllZ/2429-local-cache-implementation)

**Pull request merged**